### DISCUSSION TITLE
Setup a Dev/Test instance with DB replication from Prod

### DISCUSSION THREAD

**Discussion by Jaspotn0612** [snipe/snipe-it#8972] [Show and tell]
I think this might be a relevant discussion topic and right category so sorry if not. I did post this as a query (https://github.com/snipe/snipe-it/issues/8269) but haven't gained any replies so trying to cast a wider net.

My company has a prod instance and also a test instance so we can test upgrades or have a sandpit before we do something in our prod setup. In any logical sense, it's the way to go.

However, the issue is a common one with a sandpit, at some point you have played about too much and need to again realign your Dev/Test with prod so you ensure things are ok. Yes I know you normally do stuff in Test -> Dev -> Prod and if something isn't going in that flow you should revert but it's not a perfect system and eventually, the Test asset list has drifted too far from prod that data isn't' there as most updates are in prod.

So I'm wondering how others (if any) have been able to set up a Prod -> Dev/Test replication so that a restore can be done. Do you simply take the latest backup of prod, do a restore and then spend half a day going through checks to make sure any prod elements are changed to Dev? What problems have you encountered with semi-scheduled restores that you've had to make a note to change each time?

**Comment by Jaspotn0612** [comment]
Ha, as I was doing this post, I see @snipe had replied to my query!.... Am still interested to find out if others have come up with a method, bearing in mind it might contain their own brand of madness...

**Reply by yosiasz** [reply]
automation is the key
Working on following madness approach using a cron job
1. backup prod every x 
2. copy to dev/test
3. restore

or are you looking for full on replication ?

**Reply by Jaspotn0612** [reply]
Pretty much what you've described is what I want, a backup of prod, put into dev and then only the relevant portions changed to signify it's a dev system such as a colour scheme, so people know what system their in, and any notification emails stating it's coming from Snipe Dev instead of Snipe Prod. More than likely we'll just disable emails from Dev to prevent users getting confused. trying to stick to the KISS method.
